velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -567,24 +567,16 @@ void ReaderBase::scheduleRowGroups(
     const std::vector<uint32_t>& rowGroupIds,
     int32_t currentGroup,
     StructColumnReader& reader) {
-  auto thisGroup = rowGroupIds[currentGroup];
-  auto nextGroup =
-      currentGroup + 1 < rowGroupIds.size() ? rowGroupIds[currentGroup + 1] : 0;
-  auto input = inputs_[thisGroup].get();
-  if (!input) {
-    inputs_[thisGroup] = reader.loadRowGroup(thisGroup, input_);
-  }
-  for (auto counter = 0; counter < options_.prefetchRowGroups(); ++counter) {
-    if (nextGroup) {
-      if (inputs_.count(nextGroup) == 0) {
-        inputs_[nextGroup] = reader.loadRowGroup(nextGroup, input_);
-      }
-    } else {
-      break;
+  auto numRowGroupsToLoad = std::min(
+      options_.prefetchRowGroups() + 1,
+      static_cast<int64_t>(rowGroupIds.size() - currentGroup));
+  for (auto i = 0; i < numRowGroupsToLoad; i++) {
+    auto thisGroup = rowGroupIds[currentGroup + i];
+    if (!inputs_[thisGroup]) {
```

Comment:
Using `count()` is more efficient than indexing.

Reply:
@majetideepak Can you point me to any articles or experiments that count() is more efficient than operator[]? If might be the case if there is no need to actually insert into inputs_ (unordered_map), but for this case we need to insert anyways.

Reply:
For both cases, `count()` is better since it does not have to return a reference to the value.
If we always insert, then there is no need to check for its existence. We should remove the if condition.
But I think it is the case here that there is no need to insert if the prefetch happens on the I/O thread.

Reply:
> If we always insert, then there is no need to check for its existence. We should remove the if condition.

loadRowgroup() is an expensive operation and should be only called if the RG is not already loaded. The if check is necessary.

After all, we are talking about a single small hashtable lookup operation that happens once per row group, and I don't think using count() or operator[] would cause any difference to the reader performance. I think the readability prevails here. `if (!inputs_[thisGroup])` is easier to understand than `if (inputs_.count(thisGroup) == 0)` and therefore I chose the former. 

But still, if you strongly believe using count() would make a difference, I'll be happy to see some microbenchmark results. But I really don't think it's worth the time.